process/api.py

```diff
@@ -87,6 +99,161 @@ class CollectionViewSet(viewsets.ViewSetMixin, ListAPIView):
         "completed_at",
     ]
 
+    @csrf_exempt
+    @action(methods=["post"], detail=False)
+    def new(self, request):
+        input_message = request.data
+        if "source_id" not in input_message or "data_version" not in input_message:
+            return Response(
+                'Unable to parse input. Please provide {"source_id": "<source_id>", "data_version": "<data_version>"}',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        if not settings.ENABLE_CHECKER and input_message.get("check"):
+            logger.error("Checker is disabled in settings - see ENABLE_CHECKER value.")
+
+        collection, upgraded_collection, compiled_collection = create_collections(
+            # Identification
+            input_message["source_id"],
+            input_message["data_version"],
+            sample=input_message.get("sample", False),
+            # Steps
+            upgrade=input_message.get("upgrade", False),
+            compile=input_message.get("compile", False),
+            check=input_message.get("check", False),
+            # Other
+            scrapyd_job=input_message.get("job", ""),
+            note=input_message.get("note"),
+        )
+
+        result = {"collection_id": collection.pk}
+        if upgraded_collection:
+            result["upgraded_collection_id"] = upgraded_collection.pk
+        if compiled_collection:
+            result["compiled_collection_id"] = compiled_collection.pk
+
+        return Response(result)
+
+    @csrf_exempt
+    @action(methods=["post"], detail=False)
+    def close(self, request):
+        input_message = request.data
```

Comment:
Like with `create`, we can use a serializer to validate stats and reason (collection_id will be in path).